src/main/java/no/nav/pto/veilarbportefolje/arenaaktiviteter/GruppeAktivitetRepository.java

```diff
@@ -0,0 +1,170 @@
+        );
+    }
+
+    public List<GruppeAktivitetSchedueldDTO> hentUtgatteAktivteter() {
+        String sql = "SELECT * FROM " + TABLE_NAME
+                + " WHERE " + MOTEPLAN_SLUTTDATO + " < CURRENT_TIMESTAMP";
+        return db.queryForList(sql)
+                .stream()
+                .map(result -> (ResultSet) result)
+                .map(this::mapTilDto)
+                .collect(toList());
+    }
+
+    public void utledOgLagreGruppeaktiviteter(PersonId personId, AktorId aktorId) {
+        List<GruppeAktivitetSchedueldDTO> gruppeAktiviteter = hentAktiveAktivteter(aktorId);
+        Timestamp nesteStart = gruppeAktiviteter.stream()
+                .filter(GruppeAktivitetSchedueldDTO::isAktiv)
+                .map(GruppeAktivitetSchedueldDTO::getAktivitetperiodeFra)
+                .min(Comparator.naturalOrder())
+                .orElse(null);
+        Timestamp nesteUtlopsdato = gruppeAktiviteter.stream()
+                .filter(GruppeAktivitetSchedueldDTO::isAktiv)
+                .map(GruppeAktivitetSchedueldDTO::getAktivitetperiodeTil)
+                .max(Comparator.naturalOrder())
+                .orElse(null);
+
+        boolean aktiv = (nesteStart != null && nesteUtlopsdato != null);
+        AktivitetStatus aktivitetStatus = new AktivitetStatus()
+                .setAktivitetType(AktivitetTyper.gruppeaktivitet.name())
+                .setAktiv(aktiv)
+                .setAktoerid(aktorId)
+                .setPersonid(personId)
+                .setNesteStart(nesteStart)
+                .setNesteUtlop(nesteUtlopsdato);
+        aktivitetDAO.upsertAktivitetStatus(aktivitetStatus);
+    }
+
+    public List<GruppeAktivitetSchedueldDTO> hentAktiveAktivteter(AktorId aktorId) {
+        String sql = "SELECT * FROM " + TABLE_NAME
+                + " WHERE " + AKTIV + " = 'J' AND " + AKTOERID + " = ?";
+        return db.queryForList(sql, aktorId.get())
+                .stream()
+                .map(this::mapTilDto)
+                .collect(toList());
+    }
+
+    @SneakyThrows
+    private GruppeAktivitetSchedueldDTO mapTilDto(ResultSet rs) {
+        boolean aktiv = "J".equals(rs.getString(AKTIV));
+        return new GruppeAktivitetSchedueldDTO()
+                .setVeiledningdeltakerId(rs.getString(VEILEDNINGDELTAKER_ID))
+                .setAktivitetperiodeFra(rs.getTimestamp(MOTEPLAN_STARTDATO))
+                .setAktivitetperiodeTil(rs.getTimestamp(MOTEPLAN_SLUTTDATO))
+                .setMoteplanId(rs.getString(MOTEPLAN_ID))
+                .setHendelseId(rs.getLong(HENDELSE_ID))
+                .setAktorId(AktorId.of(AKTOERID))
+                .setAktiv(aktiv);
+    }
+
+    @SneakyThrows
+    private GruppeAktivitetSchedueldDTO mapTilDto(Map<String, Object> rs) {
+        boolean aktiv = "J".equals(rs.get(AKTIV));
+        long hendelse = (rs.get(HENDELSE_ID) instanceof BigDecimal) ? ((BigDecimal) rs.get(HENDELSE_ID)).longValue() :(Integer) rs.get(HENDELSE_ID);
```

Comment:
interesting line :)

Reply:
But it is not good, but I was really struggling so this is what I ended up with 😓